benchmarks/reduce_by_key/cub_reduce_by_key_bench.cu

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cub/cub.cuh>
+#include <key_generator.hpp>
+#include <nvbench/nvbench.cuh>
+#include <thrust/device_vector.h>
+
+/**
+ * @brief A benchmark evaluating CUB's reduce-by-key performance.
+ */
+template <typename Key, typename Value>
+void nvbench_cub_reduce_by_key(nvbench::state& state, nvbench::type_list<Key, Value>)
```

Comment:
Not sure we need to include the cub benchmark in cuco. If the performance result is shown in the notebook, keeping the benchmark source file seems a bit redundant.

Reply:
I think it's nice to have an up-to-date view if cub changes its RBK implementation at some point or if we want to compare the numbers for a specific architecture.

Reply:
Agreed. I'd like to keep it for regenerating baselines